services/analytics/analyticsServices.ts

```diff
@@ -0,0 +1,18 @@
+'use client'
+
+import { BetterSocialEventTracking } from "analytics/analyticsEventTracking"
+
+const sendAnalytics = (event: BetterSocialEventTracking, data?: object) => {
+  fetch('/api/send-analytics', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      event,
+      data
+    })
+  })
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendAnalytics` function is currently not handling any potential errors that might occur during the fetch operation. This could lead to unhandled promise rejections if the fetch fails for any reason (e.g., network issues, server errors). It's a good practice to always handle errors in promises.

Here's how you can add error handling to this function:

```diff
  const sendAnalytics = (event: BetterSocialEventTracking, data?: object) => {
    fetch('/api/send-analytics', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        event,
        data
      })
-   })
+   }).catch(error => {
+     console.error('Error sending analytics:', error);
+   });
  }
```

This will catch any errors that occur during the fetch and log them to the console. Depending on your application's needs, you may want to handle these errors differently (e.g., by showing an error message to the user or reporting the error to an error tracking service).

Also, it would be better to return the Promise from the `sendAnalytics` function. This allows the caller to wait for the asynchronous operation to complete if necessary.

```diff
  const sendAnalytics = (event: BetterSocialEventTracking, data?: object) => {
-   fetch('/api/send-analytics', {
+   return fetch('/api/send-analytics', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        event,
        data
      })
    }).catch(error => {
      console.error('Error sending analytics:', error);
    });
  }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->